src/domain/community/user/user.service.ts

```diff
@@ -28,46 +32,39 @@ export class UserService {
     @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
   ) {}
 
-  async createUser(userData: UserInput): Promise<IUser> {
+  async createUser(userData: CreateUserInput): Promise<IUser> {
     await this.validateUserProfileCreationRequest(userData);
 
     // Ok to create a new user + save
     const user = User.create(userData);
-    await this.initialiseMembers(user);
+    await this.initialiseMembers(user, userData);
     // Need to save to get the object identifiers assigned
-    await this.userRepository.save(user);
-    this.logger.verbose?.(
-      `Created a new user with id: ${user.id}`,
-      LogContext.COMMUNITY
-    );
-
-    // Now update the profile if needed
-    const profileData = userData.profileData;
-    if (profileData && user.profile) {
-      await this.profileService.updateProfile(user.profile.id, profileData);
-    }
-    // reload the user to get it populated
-    const populatedUser = await this.getUserByIdOrFail(user.id);
-
+    const savedUser = await this.userRepository.save(user);
     this.logger.verbose?.(
-      `User ${userData.email} was created!`,
+      `Created a new user with id: ${savedUser.id}`,
       LogContext.COMMUNITY
     );
 
-    return populatedUser;
+    return savedUser;
   }
 
   // Helper method to ensure all members that are arrays are initialised properly.
   // Note: has to be a seprate call due to restrictions from ORM.
-  async initialiseMembers(user: IUser): Promise<IUser> {
+  async initialiseMembers(
```

Comment:
I think it will be cleaner if you have one method where you create the user from the UserData, not create the user from the repo method outside the method and pass it next to the UserData

Reply:
Agreed - see comment above about this split not being needed.